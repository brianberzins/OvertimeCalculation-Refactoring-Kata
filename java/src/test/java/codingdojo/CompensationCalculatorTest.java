package codingdojo;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;
import org.lambda.query.Queryable;

import java.time.Duration;

public class CompensationCalculatorTest {

    @Test
    void combinations() {
        Double[] overtimes = {-1d, 8d, 12d, 20d};
        Duration[] durations =  Queryable.as(5, 10).select(Duration::ofHours).asArray();
        Boolean[] booleans = {true, false};
        CombinationApprovals.verifyAllCombinations((a, b, c, d, e, f, g) -> {
            var assignment = new Assignment(b, c);
            var briefing = new Briefing(d, e, f, g);
            return CompensationCalculator.calculateOvertime(a, assignment, briefing);
        }, overtimes, booleans, durations, booleans, booleans, booleans, booleans);
    }

}
