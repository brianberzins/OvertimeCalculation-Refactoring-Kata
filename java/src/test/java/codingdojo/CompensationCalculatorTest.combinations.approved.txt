[-1, true, PT5H, true, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, true, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT5H, false, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, true, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, true, PT10H, false, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, true, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT5H, false, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, true, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, true, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, true, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, true, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, true, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, false, true, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, false, true, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, false, false, true] => Overtime[hoursRate1=0, hoursRate2=0] 
[-1, false, PT10H, false, false, false, false] => Overtime[hoursRate1=0, hoursRate2=0] 
[8, true, PT5H, true, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, true, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT5H, false, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, true, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, true, PT10H, false, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, true, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT5H, false, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, true, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, true, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, true, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, true, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, true, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, false, true, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, false, true, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, false, false, true] => Overtime[hoursRate1=8, hoursRate2=0] 
[8, false, PT10H, false, false, false, false] => Overtime[hoursRate1=8, hoursRate2=0] 
[12, true, PT5H, true, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, true, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, true, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, true, false, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, false, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, false, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, true, false, false, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, false, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, false, true, true, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT5H, false, true, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT5H, false, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, false, false, true, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT5H, false, false, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT5H, false, false, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT10H, true, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, true, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, true, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, true, false, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, false, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, false, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, true, false, false, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, false, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, false, true, true, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT10H, false, true, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT10H, false, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, false, false, true, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, true, PT10H, false, false, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, true, PT10H, false, false, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, true, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, true, true, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, true, true, false, true] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, true, true, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, true, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, true, false, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, true, false, false, true] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, true, false, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, false, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, false, true, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, false, true, false, true] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT5H, false, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, false, false, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, false, false, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT5H, false, false, false, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, true, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, true, true, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, true, true, false, true] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT10H, true, true, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT10H, true, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, true, false, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, true, false, false, true] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT10H, true, false, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT10H, false, true, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, false, true, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, false, true, false, true] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT10H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=2] 
[12, false, PT10H, false, false, true, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, false, false, true, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, false, false, false, true] => Overtime[hoursRate1=12, hoursRate2=0] 
[12, false, PT10H, false, false, false, false] => Overtime[hoursRate1=12, hoursRate2=0] 
[20, true, PT5H, true, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, true, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, true, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, true, false, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, false, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, false, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, true, false, false, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, false, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, false, true, true, false] => Overtime[hoursRate1=10, hoursRate2=5] 
[20, true, PT5H, false, true, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=5] 
[20, true, PT5H, false, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, false, false, true, false] => Overtime[hoursRate1=10, hoursRate2=5] 
[20, true, PT5H, false, false, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT5H, false, false, false, false] => Overtime[hoursRate1=10, hoursRate2=5] 
[20, true, PT10H, true, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, true, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, true, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, true, false, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, false, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, false, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, true, false, false, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, false, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, false, true, true, false] => Overtime[hoursRate1=10, hoursRate2=6] 
[20, true, PT10H, false, true, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=6] 
[20, true, PT10H, false, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, false, false, true, false] => Overtime[hoursRate1=10, hoursRate2=6] 
[20, true, PT10H, false, false, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, true, PT10H, false, false, false, false] => Overtime[hoursRate1=10, hoursRate2=6] 
[20, false, PT5H, true, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, true, true, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, true, true, false, true] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT5H, true, true, false, false] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT5H, true, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, true, false, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, true, false, false, true] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT5H, true, false, false, false] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT5H, false, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, false, true, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, false, true, false, true] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT5H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT5H, false, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, false, false, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, false, false, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT5H, false, false, false, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, true, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, true, true, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, true, true, false, true] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT10H, true, true, false, false] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT10H, true, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, true, false, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, true, false, false, true] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT10H, true, false, false, false] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT10H, false, true, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, false, true, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, false, true, false, true] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT10H, false, true, false, false] => Overtime[hoursRate1=10, hoursRate2=10] 
[20, false, PT10H, false, false, true, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, false, false, true, false] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, false, false, false, true] => Overtime[hoursRate1=20, hoursRate2=0] 
[20, false, PT10H, false, false, false, false] => Overtime[hoursRate1=20, hoursRate2=0] 
